lib/install/phases/coredns.go

```diff
@@ -130,6 +129,28 @@ func (r *corednsExecutor) Execute(ctx context.Context) error {
 	return nil
 }
 
+func mergeUpstreamResolvers(configs ...*storage.ResolvConf) []string {
+	var upstreams []string
+	dedup := make(map[string]bool)
+	for _, config := range configs {
+		if config != nil {
```

Comment:
if you handle the `nil` case first, the code would not have to be so far to the right:
```go
if config == nil {
	continue
}
```